src/coreclr/System.Private.CoreLib/src/System/Exception.CoreCLR.cs

```diff
@@ -55,13 +55,32 @@ private IDictionary CreateDataContainer()
         [MethodImpl(MethodImplOptions.InternalCall)]
         private static extern bool IsImmutableAgileException(Exception e);
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern IRuntimeMethodInfo GetMethodFromStackTrace(object stackTrace);
+        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "ExceptionNative_GetMethodFromStackTrace")]
+        private static partial void GetMethodFromStackTrace(ObjectHandleOnStack stackTrace, ObjectHandleOnStack method);
 
         private MethodBase? GetExceptionMethodFromStackTrace()
         {
             Debug.Assert(_stackTrace != null, "_stackTrace shouldn't be null when this method is called");
-            IRuntimeMethodInfo method = GetMethodFromStackTrace(_stackTrace!);
+
+            // The _stackTrace can be either sbyte[] or Object[]. In the latter case,
```

Comment:
What is the benefit of doing this extraction in the managed code? It would be nice if the managed side could keep handling the _stacktrace as opaque and stay agnostic of the native side implementation details.

Reply:
I am removing HelperMethodFrames and our general policy has been to perform as many checks in managed as possible. I can keeps these in native if that is preferred.